bitcoin/script/miniscript.h

```diff
@@ -1408,7 +1408,11 @@ inline NodeRef<Key> DecodeScript(I& in, I last, const Ctx& ctx) {
         }
     }
     if (constructed.size() != 1) return {};
-    return constructed.front();
+    const NodeRef<Key> tl_node = constructed.front();
+    // Note that due to how ComputeType works (only assign the type to the node if the
+    // subs' types are valid) this would fail if any node of tree is badly typed.
+    if (!tl_node->IsValidTopLevel()) return {};
```

Comment:
It may be useful to also add a `if (!contructed.back()->IsValid()) return {};` inside the loop. That should quickly catch bad scripts, possibly long before full parsing is complete.